src/OrchardCore/OrchardCore.Search.Elasticsearch.Core/Services/ElasticIndexingService.cs

```diff
@@ -0,0 +1,302 @@
+                                {
+                                    latestIndexContext = new BuildIndexContext(new DocumentIndex(task.ContentItemId, contentItem.ContentItemVersionId), contentItem, new string[] { contentItem.ContentType }, new ContentIndexSettings());
+                                    await indexHandlers.InvokeAsync(x => x.BuildIndexAsync(latestIndexContext), _logger);
+                                }
+                            }
+
+                            // Update the document from the index if its lastIndexId is smaller than the current task id.
+                            foreach (var index in allIndices)
+                            {
+                                if (index.Value >= task.Id || !settingsByIndex.TryGetValue(index.Key, out var settings))
+                                {
+                                    continue;
+                                }
+
+                                var context = !settings.IndexLatest ? publishedIndexContext : latestIndexContext;
+
+                                //We index only if we actually found a content item in the database
+                                if (context == null)
+                                {
+                                    //TODO purge these content items from IndexingTask table
+                                    continue;
+                                }
+
+                                var cultureAspect = await contentManager.PopulateAspectAsync<CultureAspect>(context.ContentItem);
+                                var culture = cultureAspect.HasCulture ? cultureAspect.Culture.Name : null;
+                                var ignoreIndexedCulture = settings.Culture == "any" ? false : culture != settings.Culture;
+
+                                // Ignore if the content item content type or culture is not indexed in this index
+                                if (!settings.IndexedContentTypes.Contains(context.ContentItem.ContentType) || ignoreIndexedCulture)
+                                {
+                                    continue;
+                                }
+
+                                updatedDocumentsByIndex[index.Key].Add(context.DocumentIndex);
+                            }
+                        }
+                    }
+
+                    // Commented because we can't delete a document that is not already stored.
+                    // We can delete only after the document is committed once.
+                    // We let the ElasticIndexingContentHandler do it.
+
+                    // Delete all the existing documents
+                    //foreach (var index in updatedDocumentsByIndex)
+                    //{
+                    //    var deletedDocuments = updatedDocumentsByIndex[index.Key].Select(x => x.ContentItemId);
+
+                    //    await _indexManager.DeleteDocumentsAsync(index.Key, deletedDocuments);
+                    //}
```

Comment:
Need to try different things with this here. If the ElasticIndexingContentHandler fails to delete the Content Item then here it should remove the document in the indices. Maybe we should try without catching but I want to see alternatives first.